pdns/dnsrecords.hh

```diff
@@ -786,6 +786,28 @@ private:
  uint8_t d_eui64[8];
 };
 
+#define APL_FAMILY_IPV4 1
+#define APL_FAMILY_IPV6 2
+class APLRecordContent : public DNSRecordContent
+{
+public:
+  APLRecordContent() {};
+  static void report(void);
+  static std::shared_ptr<DNSRecordContent> make(const DNSRecord &dr, PacketReader& pr);
+  static std::shared_ptr<DNSRecordContent> make(const string& zone); // FIXME400: DNSName& zone?
+  string getZoneRepresentation(bool noDot=false) const override;
+  void toPacket(DNSPacketWriter& pw) override;
+  uint16_t getType() const override { return QType::APL; }
+private:
+  uint16_t d_family;
+  uint8_t d_prefix;
+  unsigned int d_n : 1;
+  unsigned int d_afdlength : 7;
```

Comment:
Hi, for the time being (and hopefully going forward), pdns master is running the [check-spelling](https://github.com/marketplace/actions/check-spelling) action. In order to avoid a ❌ when merged https://github.com/jsoref/pdns/commit/81537223e3874315cc478ac4a590bea76e5653e2#commitcomment-39229828 -- you'll need to merge/rebase on `master` to get the check (and the files) -- you can accept the advice (which should add `afdlength` to `expect.txt`) -- but I think you shouldn't add `tbd`.

Reply:
I hope I did this the right way.